### PR TITLE
DEV-3084: Component boilerplate update to support managed identity

### PR DESCRIPTION
**Pull request by ianhitchcock**
## Description of the change

Changes required by the mint portal components in component boilerplate

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation or Development tools (readme, specs, tests, code formatting)

## Links

 - Jira issue number: DEV-3084
 - Process.st launch checklist: https://app.process.st/runs/DEV-3084-Component-boilerplate-update-to-support-managed-identity-vKYJqQai9cIsBizn3G1KlQ/tasks/pRTzosjsbYAU-LRpJ8hC5g

### Development

- [x] [Prettier](https://www.npmjs.com/package/prettier) was run
- [x] The behaviour changes in the pull request are covered by specs
- [x] All tests related to the changed code pass in development

### Paperwork

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has a Jira number
- [x] This pull request has a Process.st launch checklist
- [ ] "Ready for review" label attached to the PR and reviewers pinged on Slack

### Code review 

- [x] Changes have been reviewed by at least one other engineer
- [x] Security impacts of this change have been considered

### CODE REVIEW

File: packages/component-boilerplate/src/environment/environment.ts [hunk 1]
```diff
@@ -215,6 +233,15 @@ export function useAppDomain(): string {
   }
 }
 
+const DEFAULT_PORTAL_AUTH_URL = "https://portal-auth.referralsaasquatch.com"; //todo: finalize
```
Comment (johanventer) [inline packages/component-boilerplate/src/environment/environment.ts:236]: Flagging todo

File: packages/component-boilerplate/src/index.ts [hunk 2]
```diff
@@ -16,20 +16,34 @@ export {
   useProgramId,
   setProgramId,
   useAppDomain,
+  usePortalAuthUrl,
```
Comment (johanventer) [inline packages/component-boilerplate/src/index.ts:19]: Remove

**Review by johanventer — CHANGES_REQUESTED** [review]

File: packages/component-boilerplate/src/environment/environment.ts [hunk 3]
```diff
@@ -215,6 +222,15 @@ export function useAppDomain(): string {
...
+export function usePortalAuthUrl(): string {
+  const sdk = getEnvironmentSDK();
+  if (sdk.type === "SquatchPortal") {
+    return sdk.env?.portalAuthUrl || DEFAULT_PORTAL_AUTH_URL;
+  }
+  return DEFAULT_PORTAL_AUTH_URL;
+}
+
```
[the remark this answers is not in the document]
Comment (ianhitchcock) [reply packages/component-boilerplate/src/environment/environment.ts]: Removed :+1:

File: packages/component-boilerplate/src/index.ts (hunk 2 above)
Comment (ianhitchcock) [reply packages/component-boilerplate/src/index.ts]: Removed :+1:

**Review by johanventer — DISMISSED** [review]

**Review by johanventer — APPROVED** [review]

**Pull request merged**